### PR TITLE
Experimental - Risk - BucketAnalysis

### PR DESCRIPTION
**Pull request by rimrockcapital**
Added stand alone function bucketAnalysis, along with corresponding unit test.

### CODE REVIEW

File: quantlib/experimental/risk/sensitivityanalysis.pyx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+from quantlib.instruments.instrument cimport Instrument
+cimport quantlib.instruments.bonds as _bo
+
+
+cdef public enum SensitivityAnalysis:
+    OneSide
+    Centered
+ 
+
+def bucketAnalysis(quotes_vvsq, instruments,
```
Comment (dpinte) [inline quantlib/experimental/risk/sensitivityanalysis.pyx]: pep8 would recommend `bucket_analysis`

File: quantlib/experimental/risk/sensitivityanalysis.pyx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                    quantity, shift, type):
+
+    """ 
+		Inputs:
```
Comment (dpinte) [inline quantlib/experimental/risk/sensitivityanalysis.pyx]: Can you update the format of the docstring to match the standards?

File: quantlib/experimental/risk/sensitivityanalysis.pyx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+cimport quantlib.instruments.bonds as _bo
+
+
+cdef public enum SensitivityAnalysis:
+    OneSide
+    Centered
+ 
+
+def bucketAnalysis(quotes_vvsq, instruments,
+                    quantity, shift, type):
```
Comment (dpinte) [inline quantlib/experimental/risk/sensitivityanalysis.pyx]: type is a reserved word, you should use `type_` or a name closer to what that type really means, e.g. `analysis_type`

**Comment by dpinte** [comment]
@rimrockcapital some styling comments but nothing major.

**Comment by rimrockcapital** [comment]
This is the last major PR for a while. 
Made the suggested changes. Anything else I need to do to merge ?

**Comment by dpinte** [comment]
@rimrockcapital I'll review that asap. This sounds great!

**Pull request merged**